### PR TITLE
Make order_column option more flexible

### PR DESCRIPTION
**Pull request by Kulgar**
Hello everyone, 

I'm doing this little pull request to address a limitation I faced using the gem. 

Indeed, I have a "jsonb" column that stores translations like this in my model "Tag" that uses this gem: 
`label_i18n: { en: "Label", fr: "Sujet" }`

I wanted to sort the children of tags like this: 

`SELECT "tags".* FROM "tags" ORDER BY ("tags"."label_i18n" ->> 'fr') ASC`

But I can't do that with the current options as you are forcing the "=> :asc" part. 

So... as the "=> :asc" is implicit when you don't specify it, I was just thinking: why not removing it so that the order_column option can accept more complexe options? 

I added a spec to test a more complexe option: `{ :name => :desc }` which is a more common usecase than mine I think. 

Thanks in advance for any feedbacks!

### CODE REVIEW

**Comment by Kulgar** [comment]
No feedbacks on my PR?

**Review by parndt — APPROVED** [review]

File: spec/awesome_nested_set_spec.rb [hunk 1]
```diff
@@ -1225,6 +1225,13 @@ def check_structure(entries, structure)
     it "should sort by custom sort column" do
       expect(OrderedCategory.acts_as_nested_set_options[:order_column]).to eq('name')
       expect(OrderedCategory.order_column_name).to eq('name')
+      expect(OrderedCategory.first.children.explain).to include('ORDER BY name')
+    end
+
+    it "should sort by custom hash sort" do
+      expect(HashOrderedCategory.acts_as_nested_set_options[:order_column]).to eq({ :name => :desc })
+      expect(HashOrderedCategory.order_column_name).to eq({ :name => :desc })
+      expect(HashOrderedCategory.first.children.explain).to include('ORDER BY "categories"."name" DESC')
```
Comment (parndt) [inline spec/awesome_nested_set_spec.rb:1234]:
```suggestion
      expect(HashOrderedCategory.first.children.explain).to include(
        "ORDER BY #{HashOrderedCategory.quoted_table_name}.#{HashOrderedCategory.connection.quote_column_name(:name)} DESC"
      )
```

**Comment by parndt** [comment]
Thanks @Kulgar it's great 😄

**Comment by Kulgar** [comment]
@parndt : thank you for reviewing and accepting my PR :-)

**Pull request merged**